src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -318,24 +370,67 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
         }
     }
 
-    private long writePQ(SequentialWriter writer) throws IOException
+    // taken and adopted from com.github.jbellis.jvector.disk.OnDiskGraphIndex
+    private static <T> void writeGraph(GraphIndex<T> graph,
+                                       RandomAccessVectorValues<T> vectors,
+                                       DataOutput out,
+                                       Function<Integer, Integer> ordinalMapper) throws IOException {
+        assert graph.size() == vectors.size() : "graph size " + graph.size() + " != vectors size " + vectors.size();
+
+        GraphIndex.View<T> view = graph.getView();
+        out.writeInt(graph.size());
+        out.writeInt(vectors.dimension());
+        out.writeInt(ordinalMapper.apply(view.entryNode()));
+        out.writeInt(graph.maxEdgesPerNode());
+
+        for(int node = 0; node < graph.size(); ++node) {
+            out.writeInt(node);
+            int originalNode = ordinalMapper.apply(node);
+            Io.writeFloats(out, (float[])vectors.vectorValue(originalNode));
+            NodesIterator neighbors = view.getNeighborsIterator(originalNode);
+            out.writeInt(neighbors.size());
+
+            int n;
+            for(n = 0; n < neighbors.size(); ++n) {
+                out.writeInt(ordinalMapper.apply(neighbors.nextInt()));
+            }
+
+            assert !neighbors.hasNext();
+
+            while(n < graph.maxEdgesPerNode()) {
+                out.writeInt(-1);
+                ++n;
+            }
+        }
+    }
+
+    private long writePQ(SequentialWriter writer, Function<Integer, Integer> ordinalMapper) throws IOException
     {
         // don't bother with PQ if there are fewer than 1K vectors
         int M = vectorValues.dimension() / 2;
         writer.writeBoolean(vectorValues.size() >= 1024);
         if (vectorValues.size() < 1024)
         {
-            logger.debug("Skipping PQ for only {} vectors", vectorValues.size());
+            if (logger.isDebugEnabled())
+            {
```

Comment:
nit: generally speaking, since C* style is brace-on-newline, my preference as the lesser evil is to leave single-line statements unbraced